### PR TITLE
#54879 Docblock corrections for `get_block_file_template()`

### PR DESCRIPTION
**Pull request by johnbillion**
Trac ticket: https://core.trac.wordpress.org/ticket/54879

~~Adds a missing `@since` tag for the `block_core_navigation_render_fallback` filter too.~~

### CODE REVIEW

**Review by SergeyBiryukov — APPROVED** [review]

File: src/wp-includes/block-template-utils.php [hunk 1]
```diff
@@ -810,27 +810,27 @@ function get_block_template( $id, $template_type = 'wp_template' ) {
 }
 
 /**
- * Retrieves a single unified template object using its id.
+ * Retrieves an array of template objects by id.
```
Comment (costdev) [inline src/wp-includes/block-template-utils.php:813]: Running this outputs a single WP_Block_Template object:

```php
get_block_file_template( 'twentytwentytwo//page' )
```
@johnbillion Can you clarify "an array of template objects by id" for me?

File: src/wp-includes/block-template-utils.php [hunk 2]
```diff
@@ -810,27 +810,27 @@ function get_block_template( $id, $template_type = 'wp_template' ) {
...
  *
  * @since 5.9.0
  *
  * @param string $id            Template unique identifier (example: theme_slug//template_slug).
  * @param array  $template_type Optional. Template type: `'wp_template'` or '`wp_template_part'`.
```
Comment (costdev) [inline src/wp-includes/block-template-utils.php]:
```suggestion
 * @param string  $template_type Optional. Template type: `'wp_template'` or `'wp_template_part'`.
```

Shouldn't `$template_type` be `string`? It's passed to `_get_block_template_file()` which reads:

```
@param string $template_type 'wp_template' or 'wp_template_part'.
```

If so, then I believe the same change is also needed in:
-  Filters `pre_get_block_file_template` and `get_block_file_template`.
- `get_block_template()` + Filters `pre_get_block_template` and `get_block_template`.
- `get_block_templates()` + Filters `pre_get_block_templates` and `get_block_templates`.
- `_build_block_template_result_from_file()`.

**Review by costdev — CHANGES_REQUESTED** [review]

File: src/wp-includes/block-template-utils.php (hunk 2 above)
Comment (SergeyBiryukov) [reply src/wp-includes/block-template-utils.php]: Good catch!

File: src/wp-includes/block-template-utils.php (hunk 1 above)
Comment (johnbillion) [reply src/wp-includes/block-template-utils.php:813]: This is indeed not right. The types throughout the docs are incorrect.

**Review by audrasjb — APPROVED** [review]
This changeset looks good to go 👍

**Comment by audrasjb** [comment]
Committed in https://core.trac.wordpress.org/changeset/52631

**Pull request closed without merging**